sway-fmt-v2/src/utils/expr/mod.rs

```diff
@@ -0,0 +1,953 @@
+    name.format(formatted_code, formatter)?;
+    if let Some(contract_args) = &contract_args_opt {
+        ExprStructField::open_curly_brace(formatted_code, formatter)?;
+        let contract_args = &contract_args.clone().into_inner();
+        match formatter.shape.code_line.line_style {
+            LineStyle::Inline => {
+                contract_args.format(formatted_code, formatter)?;
+            }
+            _ => {
+                contract_args.format(formatted_code, formatter)?;
+            }
+        }
+        ExprStructField::close_curly_brace(formatted_code, formatter)?;
+    }
+    Expr::open_parenthesis(formatted_code, formatter)?;
+    formatter.shape.reset_line_settings();
+    args.clone()
+        .into_inner()
+        .format(formatted_code, formatter)?;
+    Expr::close_parenthesis(formatted_code, formatter)?;
+
+    Ok(())
+}
+
+fn get_field_width(
+    fields: &Punctuated<ExprStructField, CommaToken>,
+    formatter: &mut Formatter,
+) -> Result<(usize, usize), FormatterError> {
+    let mut largest_field: usize = 0;
+    let mut body_width: usize = 3; // this is taking into account the opening brace, the following space and the ending brace.
+    for (field, comma_token) in &fields.value_separator_pairs {
+        let mut field_length = field.field_name.as_str().chars().count() as usize;
+        if let Some((colon_token, expr)) = &field.expr_opt {
+            let mut buf = String::new();
+            write!(buf, "{} ", colon_token.span().as_str())?;
+            expr.format(&mut buf, formatter)?;
+            field_length += buf.chars().count() as usize;
+        }
+        field_length += comma_token.span().as_str().chars().count() as usize;
+        body_width += &field_length + 1; // accounting for the following space
+
+        if field_length > largest_field {
+            largest_field = field_length;
+        }
+    }
+    if let Some(final_value) = &fields.final_value_opt {
+        let mut field_length = final_value.field_name.as_str().chars().count() as usize;
+        if let Some((colon_token, expr)) = &final_value.expr_opt {
+            let mut buf = String::new();
+            write!(buf, "{} ", colon_token.span().as_str())?;
+            expr.format(&mut buf, formatter)?;
+            field_length += buf.chars().count() as usize;
+        }
+        body_width += &field_length + 1; // accounting for the following space
+
+        if field_length > largest_field {
+            largest_field = field_length;
+        }
+    }
+
+    Ok((largest_field, body_width))
+}
+
+// Leaf Spans
+
+// TODO: Find a better way of handling Boxed version
+impl LeafSpans for Box<Expr> {
+    fn leaf_spans(&self) -> Vec<ByteSpan> {
+        visit_expr(self)
+    }
+}
+
+impl LeafSpans for Expr {
+    fn leaf_spans(&self) -> Vec<ByteSpan> {
+        visit_expr(self)
+    }
+}
+
+/// Collects various expr field's ByteSpans.
+fn visit_expr(expr: &Expr) -> Vec<ByteSpan> {
```

Comment:
This function might be better named `expr_leaf_spans` as we are just matching on the expr to collect leaf spans, not really doing visitor traversal or anything like that.